packages/zuora-billing-adapter/src/filters/workflow_and_task_references.ts

```diff
@@ -80,14 +79,95 @@ const addTaskDependencies = (
       // CUSTOM_OBJECT_SUFFIX appended for lookup
       const fieldId = fieldLowercaseLookup.get(typeName.toLowerCase(), fieldName)
       if (fieldId !== undefined) {
-        return [new ReferenceExpression(fieldId)]
+        return [{
+          reference: new ReferenceExpression(fieldId),
+          location: new ReferenceExpression(
+            parametersFieldsElemId.createNestedID(typeName, fieldName)
+          ),
+        }]
       }
       return []
     })
   })
+}
+
+const addStringsReferencesDependency = (
+  inst: InstanceElement,
+  fieldLowercaseLookup: multiIndex.Index<[string, string], ElemID>,
+  parentWorkflow?: InstanceElement
+): FlatDetailedDependency[] => {
+  const dependencies: FlatDetailedDependency[] = []
+  const func: WalkOnFunc = ({ value, path }) => {
+    if (!_.isString(value)) {
+      return WALK_NEXT_STEP.RECURSE
+    }
+
+    const potentialReferencesStrings = value.match(TASK_REFS_REGEX)
+    if (potentialReferencesStrings === null || _.isEmpty(potentialReferencesStrings)) {
+      return WALK_NEXT_STEP.SKIP
+    }
+
+    const potentialReferencesNameParts = _.uniq(potentialReferencesStrings).map(str => {
+      const parts = str.split(ElemID.NAMESPACE_SEPARATOR)
+      return parts.slice(-2)
+    })
+
+    const references = potentialReferencesNameParts.map(([typeName, fieldName]) => {
+      const fieldId = fieldLowercaseLookup.get(typeName.toLowerCase(), fieldName)
+      if (isDefined(fieldId)) {
+        return new ReferenceExpression(fieldId)
+      }
 
+      if (typeName !== WORKFLOW_PARAMS_REF || !isDefined(parentWorkflow)) {
+        return undefined
+      }
+
+      const workflowParams = resolvePath(
+        parentWorkflow, parentWorkflow.elemID.createNestedID(...WORKFLOW_PARAMS_PATH)
+      )
+      if (!_.isArray(workflowParams)) {
+        return undefined
+      }
+
+      return workflowParams.some(field =>
+        field.object_name === WORKFLOW_PARAMS_REF && field.field_name === fieldName)
+        ? new ReferenceExpression(
+          parentWorkflow.elemID.createNestedID(...WORKFLOW_PARAMS_PATH)
+        )
+        : undefined
```

Comment:
readability nit (given the rest of this function's flow):
```suggestion
      if (workflowParams.some(field =>
        field.object_name === WORKFLOW_PARAMS_REF && field.field_name === fieldName
      )) {
        return new ReferenceExpression(
          parentWorkflow.elemID.createNestedID(...WORKFLOW_PARAMS_PATH)
        )
      return undefined
```